tests/data/test_dataset_splits.py

```diff
@@ -0,0 +1,242 @@
+                    203321: "test",
+                    199617: "test",
+                    246253: "test",
+                    253429: "test",
+                    254084: "train",
+                    77399: "train",
+                    62302: "train",
+                },
+            },
+        ),
+    ],
+)
+def test_dataset_splits(
+    remove_unused_data: bool,
+    split_type: str,
+    training_size: int,
+    validation_size: int,
+    expected_splits: dict[str, Any],
+) -> None:
+    """
+    Test the dataset splitter with a small test dataset.
+
+    Parameters
+    ----------
+    remove_unused_data : bool
+        Indicating whether unused metadata should be removed.
+    split_type : str
+        Type of split to be performed.
+    training_size
+        Size of the training set.
+    validation_size
+        Size of the validation set.
+    expected_splits : dict[str, Any]
+        Dictionary of expected splits.
+    """
+    test_data_path = (
+        Path(PAINT_ROOT) / "tests" / "data" / "test_data" / "test_metadata.csv"
+    )
+    expected_splits_df = pd.DataFrame(expected_splits)
+    with tempfile.TemporaryDirectory() as temp_dir:
+        output_dir = temp_dir
+        splitter = DatasetSplitter(
+            input_file=test_data_path,
+            output_dir=output_dir,
+            remove_unused_data=remove_unused_data,
+        )
+        split = splitter.get_dataset_splits(
+            split_type=split_type,
+            training_size=training_size,
+            validation_size=validation_size,
+        )
+
+        assert split.equals(expected_splits_df)
+
+
+@pytest.mark.parametrize(
+    "split_type, training_size, validation_size",
+    [
+        ("azimuth", 3, 10),
+        ("solstice", 10, 1),
+        ("this_does_not_exist", 1, 1),
+    ],
+)
+def test_make_dataset_splitter_fail(
+    split_type: str, training_size: int, validation_size: int
+) -> None:
+    """
+    Make the dataset splitter fail.
+
+    This test either provides training and validation sizes resulting in an empty list of heliostats or a split type
+    that is not valid. All of these combinations lead to failure.
+
+    Parameters
+    ----------
+    split_type : str
+        Type of split to be performed.
+    training_size
+        Size of the training set.
+    validation_size
+        Size of the validation set.
+    """
+    test_data_path = (
+        Path(PAINT_ROOT) / "tests" / "data" / "test_data" / "test_metadata.csv"
+    )
+    with tempfile.TemporaryDirectory() as temp_dir:
+        output_dir = temp_dir
+        splitter = DatasetSplitter(
+            input_file=test_data_path,
+            output_dir=output_dir,
+        )
+        with pytest.raises(ValueError):
+            splitter.get_dataset_splits(
+                split_type=split_type,
+                training_size=training_size,
+                validation_size=validation_size,
+            )
+
+
+def test_make_solstice_distance_fail():
```

Comment:
I'd suggest making a similar function to test one wrong string for `split_type` here to make sure the else block with the exception is also covered by the tests.

Reply:
This is covered by the test case in line 198. The else block can never be reached due to ``allowed_strings`` being checked earlier. Therefore, the else block has been removed!